articles/hibernate-one-to-one-mapping/index.md

```diff
@@ -0,0 +1,535 @@
+    }
+
+    public void setPrice(Double price) {
+        this.price = price;
+    }
+
+    public String getDescrition() {
+        return descrition;
+    }
+
+    public void setDescrition(String descrition) {
+        this.descrition = descrition;
+    }
+
+    public Customer getCustomer() {
+        return customer;
+    }
+
+    public void setCustomer(Customer customer) {
+        this.customer = customer;
+    }
+
+    @Override
+    public String toString() {
+        return "Package{" +
+                "id=" + id +
+                ", name='" + name + '\'' +
+                ", price=" + price +
+                ", descrition='" + descrition + '\'' +
+                '}';
+    }
+}
+ ```
+ The CascadeType.REMOVE has been left out to avoid deleting the customer once a subscription is deleted to demonstrate cascading in action.
+
+mappedBy = "subscription" - added to mark the relationship as bi-directional meaning when a subscription is loaded the customer can be identified and vice versa.
+
+## Step 4: Creating and executing the main app
+---
+In this stage, the following will be covered.
+- Creating a new customer.
+- Retrieving customer by id and assigning a new subscription to the customer using a setter method.
+- Retrieving customer by id and deleting to see cascade delete applied to the subscription.
+- Retrieve and update a customer using a setter method.
+
+##### create a new customer
+---
+```java
+import com.javadev.hibernate.demo.entity.Customer;
+import com.javadev.hibernate.demo.entity.Subscription;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.cfg.Configuration;
+
+
+public class CreateCustomer {
+    public static void main(String[] args) {
+        //create session factory
+        SessionFactory factory = new Configuration()
+                .configure("hibernate.cfg.xml")
+                .addAnnotatedClass(Customer.class)
+                .addAnnotatedClass(Subscription.class)
+                .buildSessionFactory();
+        //create session
+        Session session = factory.getCurrentSession();
+        try {
+            //create the object
+            Customer tempCustomer = new Customer("john","doe","john@javadev.com");
+
+            //start a transaction
+            session.beginTransaction();
+            //save the object
+            System.out.println("Saving the customer");
+            session.persist(tempCustomer);
+
+            //commit the transaction
+            session.getTransaction().commit();
+            System.out.println("Done!!");
+        }finally {
+            //clean up code
+            session.close();
+            factory.close();
+        }
+
+    }
+}
+```
+When the main app is executed table customer and subscription will be created and a new row will be inserted in the customer table, use the following command to see the created tables and inserted row.
+
+```SQL
+use database_name;
+
+show tables;
+
+select * from the customer;
```

Comment:
should be customer not "the customer" as you are accessing the "customer" table not the "the customer" table